xds/internal/balancer/cdsbalancer/cdsbalancer.go

```diff
@@ -609,21 +609,7 @@ func (b *cdsBalancer) generateDMsForCluster(name string, depth int, dms []cluste
 			Cluster:               cluster.ClusterName,
 			EDSServiceName:        cluster.EDSServiceName,
 			MaxConcurrentRequests: cluster.MaxRequests,
-		}
-		if cluster.LRSServerConfig == xdsresource.ClusterLRSServerSelf {
-			bootstrapConfig := b.xdsClient.BootstrapConfig()
-			parsedName := xdsresource.ParseName(cluster.ClusterName)
-			if parsedName.Scheme == xdsresource.FederationScheme {
-				// Is a federation resource name, find the corresponding
-				// authority server config.
-				if cfg, ok := bootstrapConfig.Authorities[parsedName.Authority]; ok {
-					dm.LoadReportingServer = cfg.XDSServer
-				}
-			} else {
-				// Not a federation resource name, use the default
-				// authority.
-				dm.LoadReportingServer = bootstrapConfig.XDSServer
-			}
+			LoadReportingServer:   cluster.LRSServerConfig,
```

Comment:
IIRC we're still missing something because we aren't supporting LRS for aggregate clusters?

Reply:
As mentioned in https://github.com/grpc/grpc-go/issues/7192, we currently don't support LRS for non-EDS clusters, and according to Mark, that is a bug. And with regards to aggregate clusters, Mark mentioned that the expected behavior is specified in A75, but Go hasn't implemented A74 and A75 yet.

So, in this PR, I didn't want to add LRS support for non-EDS clusters. I will do that as part of #7192.